pkg/controllers/management/feature/feature_handler.go

```diff
@@ -0,0 +1,100 @@
+package feature
+
+import (
+	"context"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	"github.com/rancher/rancher/pkg/features"
+	managementv3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/wrangler"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type handler struct {
+	featuresClient managementv3.FeatureClient
+	tokensLister   managementv3.TokenCache
+	tokenEnqueue   func(string, time.Duration)
+}
+
+func Register(ctx context.Context, wContext *wrangler.Context) {
+	h := handler{
+		featuresClient: wContext.Mgmt.Feature(),
+		tokensLister:   wContext.Mgmt.Token().Cache(),
+		tokenEnqueue:   wContext.Mgmt.Token().EnqueueAfter,
+	}
+	wContext.Mgmt.Feature().OnChange(ctx, "feature-handler", h.sync)
+}
+
+func (h *handler) sync(_ string, obj *v3.Feature) (*v3.Feature, error) {
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	obj, err := h.setLockedValue(obj)
+	if err != nil {
+		return obj, err
+	}
+
+	if obj.Name == features.TokenHashing.Name() {
+		return obj, h.refreshTokens()
+	}
+	return obj, nil
+}
+
+func (h *handler) refreshTokens() error {
+	tokenList, err := h.tokensLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, token := range tokenList {
+		if token.Labels != nil && token.Labels[tokens.TokenHashed] == "true" {
+			continue
+		}
+		h.tokenEnqueue(token.Name, 10*time.Second)
+	}
+	return nil
+}
+
+// setLockedValue evaluates whether a value should be written to the lockedValue
+// field on status and records the value if so.
+func (h *handler) setLockedValue(obj *v3.Feature) (*v3.Feature, error) {
+	lockedValue := LockedValue(obj)
+
+	if lockedValue == nil && obj.Status.LockedValue == nil {
+		return obj, nil
+	}
+	update := (lockedValue == nil && obj.Status.LockedValue != nil) || (lockedValue != nil && obj.Status.LockedValue == nil) ||
+		*lockedValue == *obj.Status.LockedValue
+
+	if !update {
+		return obj, nil
+	}
+
+	featureCopy := obj.DeepCopy()
+	featureCopy.Status.LockedValue = lockedValue
+	return h.featuresClient.Update(featureCopy)
+}
+
+// LockedValue evaluates whether updates to a feature's effective value
+// shouldl be prevented. If so LockedValue returns the value that should
+// be pinned to a feature. If nil is returned, the features value can be
+// changed and those changes should toggle the associated behavior.
+func LockedValue(obj *v3.Feature) *bool {
```

Comment:
This name is very generic and also used as a var, how about `isFeatureLocked`

Reply:
`isFeatureLocked` is not accurate. It is returning the locked value not whether it is locked. For example if it returns `false`, then the locked value is `false`. `isFeatureLocked` would imply `false` to mean that the feature is not locked.

Reply:
Add something like this in the comment
```
Return values
*  nil - not currently locked
* false - currently locked and false value
* true - currently locked and true value
```
And func rename to something like: `EvaluateLockedValueFromSpec`